apps/subscription.go

```diff
@@ -12,42 +14,62 @@ import (
+	//
+	// To receive and process these notifications as Bot, the bot must first be
+	// added as a channel member with an AddChannelMember API call.
 	SubjectUserJoinedChannel Subject = "user_joined_channel"
 	SubjectUserLeftChannel   Subject = "user_left_channel"
 
-	// SubjectBotJoinedChannel and SubjectBotLeftChannel subscribes to
-	// SubjectUserJoinedChannel and SubjectUserLeftChannel plugin events,
-	// specifically when the App's bot is added or removed from the channel
-	SubjectBotJoinedChannel Subject = "bot_joined_channel"
-	SubjectBotLeftChannel   Subject = "bot_left_channel"
-
-	// SubjectUserJoinedTeam and SubjectUserLeftTeam subscribes to respective
-	// plugin events, for the specified team. By default notifications include
-	// ActingUserID, UserID, and TeamID, but only ActingUser is fully expanded.
-	// Expand can be used to expand other entities.
+	// SubjectUserJoinedTeam, SubjectUserLeftTeam watch the specified
+	// team for users joining and leaving it.
+	//   TeamID: specifies the team to watch.
+	//   ChannelID: must be empty.
+	//   Expandable: Team, User, TeamMember.
+	//   Requires: model.PermissionViewTeam
+	//
+	// To receive and process these notifications as Bot, the bot must first be
+	// added as a channel member with an AddTeamMember API call.
 	SubjectUserJoinedTeam Subject = "user_joined_team"
 	SubjectUserLeftTeam   Subject = "user_left_team"
 
-	// SubjectBotJoinedTeam and SubjectBotLeftTeam subscribes to
-	// SubjectUserJoinedTeam and SubjectUserLeftTeam plugin events,
-	// specifically when the App's bot is added or removed from the team
+	// SubjectBotJoinedChannel, SubjectBotLeftChannel watches for the event when
+	// the app's own bot is added to, or removed from any channel in the
+	// specified team.
+	//   TeamID: specifies the team to watch.
+	//   ChannelID: must be empty, all channels are watched.
+	//   Expandable: Channel, User (will be the app's bot user), ChannelMember.
+	//   Requires: none - if the event fires, the app's bot already has the permissions.
+	SubjectBotJoinedChannel Subject = "bot_joined_channel"
+	SubjectBotLeftChannel   Subject = "bot_left_channel"
+
+	// SubjectBotJoinedTeam, SubjectBotLeftTeam system-wide watch for app's own
+	// bot added to, or removed from teams.
+	//   TeamID: must be empty.
+	//   ChannelID: must be empty.
+	//   Expandable: Team, User (will be the app's bot user), TeamMember.
+	//   Requires: none - if the event fires, the app's bot already has the permissions.
 	SubjectBotJoinedTeam Subject = "bot_joined_team"
 	SubjectBotLeftTeam   Subject = "bot_left_team"
 
-	// SubjectChannelCreated subscribes to ChannelHasBeenCreated plugin events,
-	// for the specified team. By default notifications include UserID (creator),
-	// ChannelID, and TeamID, but only Channel is fully expanded. Expand can be
-	// used to expand other entities.
+	// SubjectChannelCreated watches for new channels in the specified team.
```

Comment:
nit: could be useful to not remove "subscribes to ChannelHasBeenCreated plugin events" to keep the lineage.

Reply:
Going to remove them to match the plugin method names

Reply:
still TODO?